### DISCUSSION TITLE
Failure to install Octoeverywhere/Mobileraker Companion on Creality OS Fix

### DISCUSSION THREAD

**Discussion by EchotreePrints** [Guilouz/Creality-Helper-Script-Wiki#720] [Issues]
Greetings Programs!

I have recently encountered a couple of issues with the Helper-Script failing to install the Octoeverywhere and the Mobileraker Companion plugins. Luckily, after being dumb for several weeks, I found the answer. QuinnDamerell, told me how to fix this, but having no real experience with linux absolutely misunderstood what he was suggesting. It wasn't until I found a year old reddit thread that I actually understood it.

You will need to ssh to the machine and run a series of simple commands:

For Octoeverywhere:

~~`sudo rm -rf /usr/data/octoeverywehre-env /usr/data/octoeverywhere`~~
`sudo rm -rf /usr/data/octoeverywhere-env /usr/data/octoeverywhere`

This will remove the octoeverywhere venv and directory that are installed with the helper script.

`cd /usr/data/`
moves our active directory to the /usr/data directory.

`git clone https://github.com/QuinnDamerell/OctoPrint-OctoEverywhere.git octoeverywhere`
clones the latest version of the octoeverywhere plugin to the usr/data/octoeverywhere dir, effectively replacing the version originally installed with the helper script

`cd octoeverywhere`
moves us into the octoeverywhere dir

~~`sh isntall.sh`~~
`sh install.sh`
runs the install script to reinstall the good version.

For Mobileraker:

`sudo rm -rf /usr/data/mobileraker-env`
This will remove the mobileraker venv and directory that are installed with the helper script.

`cd /usr/data/mobileraker_companion/scripts`
moves our active directory to the /usr/data/mobileraker_companion/scripts directory.

~~`sh isntall.sh`~~
`sh install.sh`
runs the mobileraker companion installer

This worked for me as far as allowing the mobileraker plugin to be used. If it doesn't work, You'll need to follow similar steps as the octoeverywhere plugin, removing both the venv and main folder, cloning the repository, and then running the install script.

This is courtesy of Destinal from the Devil Design discord for identifying the fix, and u/predtech on reddit for posting the fix, as well as QuinnDamarell for telling me how to fix it, even if I was too ignorant to figure out what he meant lmao.

[Edit 01/19/2024]
Fixed a couple of typos in the copy/paste commands. I blame the dyslexia.

**Comment by VashBaldeus** [comment]
Great guide, meanwhile Obico ceased working due to lack of Janus being installed, wondering if compiling from source would be feasible..

**Comment by FredsMedia** [comment]
@EchotreePrints it seems like venv/python isnt even installed for me. I did a factory reset through the helper script and after folling your instructions i get this:

``` 
root@Ender3V3KE-8864` /usr/data/octoeverywhere [#] sh install.sh



      @@@@@@@@@@@@@@@@@@@@@@@@***@@@@@@@@@@@@@@@@@@@@@@@
      @@@@@@@@@@@@@@***********************@@@@@@@@@@@@@
      @@@@@@@@@@*******************************@@@@@@@@@
      @@@@@@@@***********************************@@@@@@@
      @@@@@,,,************************/////////*****@@@@
      @@@@,,,,,,*****************//////////////******@@@
      @@,,,,,,,,,,***********//////////////////*******@@
      @@,,,,,,,,,,,,*******////////****///////*********@
      @,,,,,,,,,,,/////////////////****//////***********
      @,,,,,,,//////////////////////////////************
      ,,,,,,,,////////////////////////////**************
      @,,,,,,,,,,,,/////////////////////****************
      @,,,,,,,,,,,,,,/////////////////******************
      @@,,,,,,,,,,,,,,,,//////////////*****************@
      @@@,,,,,/#######,,,,///////////*****************@@
      @@@@,,,##########,,,,,,,//////,****************@@@
      @@@@@,##########,,,,,,,,,////,,,,*************@@@@
      @@@@@########,,,,,,,,,,,,//,,,,,,,,********@@@@@@@
      @@@@@#@@@@,,,,,,,,,,,,,,,,,,,,,,,,,,,***,@@@@@@@@@
      @@@@@@@@@@@@@@@,,,,,,,,,,,,,,,,,,,,,@@@@@@@@@@@@@@

  OctoEverywhere For Klipper, Creality OS, And Bambu Lab Printers
 The 3D Printing Communities #1 Remote Access And AI Cloud Service


OctoEverywhere empowers the worldwide maker community with...
  - Free & Unlimited Mainsail, Fluidd, And Bambu Lab Printers Remote Access
  - Free & Unlimited Next-Gen AI Print Failure Detection
  - Free Full Frame Rate & Full Resolution Webcam Streaming
  - 5 Star Rated iOS & Android Apps
  - Real-Time Print Notifications
  - And So Much More


Running in K1 and K1 Max OS mode
Checking required system packages are installed...
Downloading http://bin.entware.net/mipselsf-k3.4/Packages.gz
Updated list of available packages in /opt/var/opkg-lists/entware
Package python3 (3.11.10-1) installed in root is up to date.
Package python3-pip (23.3.1-1) installed in root is up to date.
Package python3-pillow (10.1.0-1) installed in root is up to date.
Checking Python Virtual Environment For OctoEverywhere...
No virtual environment found, creating one now.
install.sh: line 208: /opt/bin/virtualenv: not found
Updating PIP if needed... (this can take a few seconds or so)
install.sh: line 294: /usr/data/octoeverywhere-env/bin/python: not found
Installing or updating required python libs...
install.sh: line 303: /usr/data/octoeverywhere-env/bin/pip3: not found
install.sh: line 434: /usr/data/octoeverywhere-env/bin/python3: not found
root@Ender3V3KE-8864 /usr/data/octoeverywhere [#] 
```

**Reply by FredsMedia** [reply]
I think i fixed it. When installing entware i didnt close the ssh session when it said to. After factory resetting and doing that it seems to be working fine.

**Reply by jktikka** [reply]
> I think i fixed it. When installing entware i didnt close the ssh session when it said to. After factory resetting and doing that it seems to be working fine.

This has been causing me a headache for days with Obico. Turns out I should have maximised the window to see the warning. Thank you sir!!

**Comment by Coolsworth** [comment]
@FredsMedia Did you ever find a fix for this? I am having the same issue with the venv

> @EchotreePrints it seems like venv/python isnt even installed for me. I did a factory reset through the helper script and after folling your instructions i get this:
> 
> ``` 
> root@Ender3V3KE-8864` /usr/data/octoeverywhere [#] sh install.sh
> 
> 
> 
>       @@@@@@@@@@@@@@@@@@@@@@@@***@@@@@@@@@@@@@@@@@@@@@@@
>       @@@@@@@@@@@@@@***********************@@@@@@@@@@@@@
>       @@@@@@@@@@*******************************@@@@@@@@@
>       @@@@@@@@***********************************@@@@@@@
>       @@@@@,,,************************/////////*****@@@@
>       @@@@,,,,,,*****************//////////////******@@@
>       @@,,,,,,,,,,***********//////////////////*******@@
>       @@,,,,,,,,,,,,*******////////****///////*********@
>       @,,,,,,,,,,,/////////////////****//////***********
>       @,,,,,,,//////////////////////////////************
>       ,,,,,,,,////////////////////////////**************
>       @,,,,,,,,,,,,/////////////////////****************
>       @,,,,,,,,,,,,,,/////////////////******************
>       @@,,,,,,,,,,,,,,,,//////////////*****************@
>       @@@,,,,,/#######,,,,///////////*****************@@
>       @@@@,,,##########,,,,,,,//////,****************@@@
>       @@@@@,##########,,,,,,,,,////,,,,*************@@@@
>       @@@@@########,,,,,,,,,,,,//,,,,,,,,********@@@@@@@
>       @@@@@#@@@@,,,,,,,,,,,,,,,,,,,,,,,,,,,***,@@@@@@@@@
>       @@@@@@@@@@@@@@@,,,,,,,,,,,,,,,,,,,,,@@@@@@@@@@@@@@
> 
>   OctoEverywhere For Klipper, Creality OS, And Bambu Lab Printers
>  The 3D Printing Communities #1 Remote Access And AI Cloud Service
> 
> 
> OctoEverywhere empowers the worldwide maker community with...
>   - Free & Unlimited Mainsail, Fluidd, And Bambu Lab Printers Remote Access
>   - Free & Unlimited Next-Gen AI Print Failure Detection
>   - Free Full Frame Rate & Full Resolution Webcam Streaming
>   - 5 Star Rated iOS & Android Apps
>   - Real-Time Print Notifications
>   - And So Much More
> 
> 
> Running in K1 and K1 Max OS mode
> Checking required system packages are installed...
> Downloading http://bin.entware.net/mipselsf-k3.4/Packages.gz
> Updated list of available packages in /opt/var/opkg-lists/entware
> Package python3 (3.11.10-1) installed in root is up to date.
> Package python3-pip (23.3.1-1) installed in root is up to date.
> Package python3-pillow (10.1.0-1) installed in root is up to date.
> Checking Python Virtual Environment For OctoEverywhere...
> No virtual environment found, creating one now.
> install.sh: line 208: /opt/bin/virtualenv: not found
> Updating PIP if needed... (this can take a few seconds or so)
> install.sh: line 294: /usr/data/octoeverywhere-env/bin/python: not found
> Installing or updating required python libs...
> install.sh: line 303: /usr/data/octoeverywhere-env/bin/pip3: not found
> install.sh: line 434: /usr/data/octoeverywhere-env/bin/python3: not found
> root@Ender3V3KE-8864 /usr/data/octoeverywhere [#] 
> ```
